connection.lua

```diff
@@ -120,6 +120,11 @@ function Connection:_error(err)
   self:_changeState(CXN_STATES.ERROR, err)
 end
 
+function Connection:_warning(err)
+  self._log(logging.WARNING, tostring(err))
+  self:_changeState(CXN_STATES.ERROR, err)
```

Comment:
In the spirit of a state machine, the distinction between ERROR and WARNING seems to be too fine grained; however, I can also see that being a source of confusion for us later. I'd prefer to avoid the state machine, but change the logged label here to something like 'ERROR/WARNING':

https://github.com/virgo-agent-toolkit/virgo-base-agent/blob/c6bf4ddfeccac434f1fe95f2f3b4eb92600ca2fb/connection.lua#L26

Reply:
@jjbuchan I pushed my proposed state label change. I'm still hesitant to introduce a new state for warning vs error.

Reply:
...and reverted that